### ISSUE TITLE
Feature Request: Support Wrapper APIs `GraphQLInteraction` and `ApolloGraphQLInteraction`

### ISSUE THREAD

**Issue by danhlee329** [pact-foundation/pact-go#129]
## Expected behaviour

I'd like to request supporting the wrapper APIs `GraphQLInteraction` and `ApolloGraphQLInteraction` that's currently available in [pact-js](https://github.com/pact-foundation/pact-js#graphql-api).  This would simplify GraphQL tests by not having to explicitly setup the HTTP request and body.

## Actual behaviour

Feature currently not supported for pact-go.

**Comment by mefellows** [comment]
Good idea - are you up for contributing? It shouldn't be a huge effort, but I haven't considered it in the Go case.

**Comment by danhlee329** [comment]
I'd be happy to contribute for this feature, I can start looking at this sometime this week.

**Comment by mefellows** [comment]
Great, thanks!

**Comment by danhlee329** [comment]
Apologize for the extra delay, will be working on this this week.

**Comment by mefellows** [comment]
No worries - thanks for the update!

**Comment by mefellows** [comment]
NOTE: GraphQL has been added in the current mainline, it needs to be ported to 2.x.x also.
